accepted/2021/preview-features/preview-features.md

```diff
@@ -0,0 +1,418 @@
+quicker to trends and having an organization and customer base that is used to
+shipping & absorbing technology quicker to make the most of today's fast moving
+world.
+
+With .NET Core, we've solved the problem of the extremely high in-place
+compatibility bar that .NET Framework has. This reduces risks for unintentional
+behavioral changes and also allows certain parts of .NET to make deliberate
+breaking changes if necessary.
+
+As a result, this makes it viable to innovate across runtime, libraries, and
+languages, allowing us to build more compelling features that would have been
+almost impossible to do on .NET Framework. Examples include static linking,
+default interface members, and ref structs (such as `Span<T>`).
+
+However, it's sometimes challenging to deliver cross-cutting feature work in a
+single release, especially when they are breaking new ground in terms of
+expressiveness. A good example of such a feature is the upcoming improved
+numeric support in generic code, with the help of static interface members. Such
+a feature requires work in the runtimes, the languages, and the libraries. The
+work at the library level isn't very high, but it's all about finding the right
+set of support infrastructure so that developers are able to be productive at
+writing generic code that can rely on numerical expressions and operators. In a
+sense, the library work can't really fully start until the runtime and compiler
+support is sufficiently bootstrapped, but it's the most important part in the
+validation of the feature set. Practically, this means a good chunk of the
+feature will only be available halfway through a release cycle, which greatly
+reduces the time window for gathering feedback from other parts of the product
+as well as from customers using preview builds of the upcoming .NET version.
+
+Sadly, preview feedback isn't always sufficient to judge how well a feature
+works because
+
+1. Many fewer people use preview builds of .NET and
+2. Many people don't use .NET previews for their production projects, thus
+   greatly reducing the breadth & depth at which previews are being exercised.
+
+This document is exploring a model where GA quality builds of .NET can contain
+features that are considered still in preview.
+
+The goal is to ensure that customers can discover those features & have an easy
+way to play with them while simultaneously being fully aware that these features
+aren't supported yet and thus might change (or be removed entirely) between
+versions of .NET, just like they could in preview builds.
+
+## Scenarios and User Experience
+
+### Enabling a preview runtime feature
+
+Maddy is building a Tye application. She reads in a blog post that the .NET
+runtime team is experimenting with a new GC mode, code named FancyGC.
+
+She wants to see how this improves the performance of her backend. She's glad to
+see that the new GC mode was shipped as part of .NET 6, so she doesn't need to
```

Comment:
In this example, the new GC mode is the default, such that there's no additional XML like `<ServerGarbageCollection>true</ServerGarbageCollection>`?

Reply:
That's good point, my example might not have been the best. Yes, the intention was we're making a change in the GC that would be on by default moving forward, as opposed to a new mode (like server vs. workstation). I guess a better example would have been tiered-JITing.

Reply:
The GC example is interesting, we are in fact considering shipping the new regions functionality within the "standalone" GC, which would mean deploying the native clrgc.dll. Wonder if enabling such capability should be captured as part of this.

Reply:
OK, this was addressed by scoping out pure runtime work.